src/stactools/osc_hazard/cubify.py

```diff
@@ -0,0 +1,155 @@
+import datetime
+import re
+import fnmatch
+import shutil
+import os.path
+from os import PathLike
+
+import numpy as np
+import xarray as xr
+import zarr
+from hazard.sources.osc_zarr import OscZarr
+
+from .constants import ATTRS
+
+INDICATOR_NAME = "indicator"
+
+
+def _parse_index_name(key: str, prefix: str = ".*"):
+    match = re.match(
+        f"{prefix}_([0-9]+\.?[0-9]+)c_([a-zA-Z0-9-]+)_([a-zA-Z0-9]+)_(\d+)*.*", key
+    )
+    if match is None:
+        raise ValueError(f"Unable to match string '{key}'")
+    temperature, gcm, scenario, year = match.groups()
+    index = dict(
+        temperature=int(temperature),
```

Comment:
```suggestion
        temperature=float(temperature),
```

Hazard now does XX.X's 🥲